### ISSUE TITLE
libspatialite Dependency with Depreicated proj_api.h

### ISSUE THREAD

**Issue by crabel99** [r-spatial/sf#1184]
I found a problem in the source code of `libspatialite` that is an issue when using PROJ 6+. As `libspatialite` is a dependency for `sf`, I thought you should be aware of the problem and the steps that I took to solve it. Because this isn't a code issue in `sf` per se, I am not writing this a pull request.

The problem centers around the precompiler flag `ACCEPT_USE_OF_DEPRECATED_PROJ_API_H` that is in `proj_api.h` in versions of PROJ 6+. 
1. Download the source http://www.gaia-gis.it/gaia-sins/libspatialite-4.3.0a.tar.gz
2. Apply the two patches from http://archive.ubuntu.com/ubuntu/pool/universe/s/spatialite/spatialite_4.3.0a-5.debian.tar.xz
3. Apply the following patch
```patch
Description: Update library to be able to compile against PROJ 6.
Author: Cal Abel <crabel@signalpowerandlight.com>

--- configure.ac
+++ configure.ac
@@ -180,7 +180,8 @@
AC_ARG_ENABLE(proj, [AS_HELP_STRING(
  [--enable-proj], [enables PROJ.4 inclusion [default=yes]])],
  [], [enable_proj=yes])
if test x"$enable_proj" != "xno"; then
-  AC_CHECK_HEADERS(proj_api.h,, [AC_MSG_ERROR([cannot find proj_api.h, bailing out])])
+  AC_DEFINE(ACCEPT_USE_OF_DEPRECATED_PROJ_API_H)
+  AC_CHECK_HEADER(proj_api.h,AC_DEFINE(HAVE_PROJ_API_H), [AC_MSG_ERROR([cannot find proj_api.h, bailing out])])
  AC_SEARCH_LIBS(pj_init_plus,proj,,AC_MSG_ERROR(['libproj' is required but it doesn't seem to be installed on this system.]),-lm)
  AC_SEARCH_LIBS(pj_init_ctx,proj,,AC_MSG_ERROR(['libproj' is required but the version on this system does not support the required features (Version 4.8.0 or later is needed).]),-lm)
--- config.h.in
+++ config.h.in
@@ -79,7 +79,10 @@
/* Define to 1 if you have the `memset' function. */
#undef HAVE_MEMSET

+/* Define to 1 to be able to use the deprecated <proj_api.h> header file. */
+#undef ACCEPT_USE_OF_DEPRECATED_PROJ_API_H
+
/* Define to 1 if you have the <proj_api.h> header file. */
#undef HAVE_PROJ_API_H

/* Define to 1 if you have the <sqlite3ext.h> header file. */
```
 **EDIT**
4. In `bash` run `aclocal && autoconf`.
Note: Compilation requires `pkg-config`

**Comment by edzer** [comment]
I don't see why libspatialite is a dependency for `sf`, other than GDAL may be built against it (or not).

**Comment by crabel99** [comment]
I didn't have to compile it when I installed `gdal`. I only got pinged that it was needed was an error code that came up during compilation. Installing the aptitude package results in the installation of the distribution version of `gdal` as a dependency which interferes with my system's configuration. This was why I had to compile it from source and make it work with PROJ 6. Just letting you know my journey!
